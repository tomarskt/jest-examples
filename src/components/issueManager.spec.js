import React from 'react';
import renderer from 'react-test-renderer';
import IssueManager from './issueManager';

jest.mock('../common/fetch');
describe('Issue Manager', () => {
    it('renders correctly', () => {
        const component = renderer.create(<IssueManager />);
        /**
        › 1 snapshot written.
            console.log src\components\issueManager.js:71
                ********** render ************

            console.log src\components\issueManager.js:71
                ********** render ************

            console.log src\components\issueManager.js:71
                ********** render ************

            Snapshot Summary
            › 1 snapshot written in 1 test suite.

            we rendered 3 times with cdm called; data fetched (via __mock__/fetch); component re-rendered
            but for some reason, snapshot is the 2nd one (loading one)
         */
        expect(component.toJSON()).toMatchSnapshot();

    })
});