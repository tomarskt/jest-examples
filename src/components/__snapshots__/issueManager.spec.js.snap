// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue Manager renders correctly 1`] = `
<div>
  <div
    id="change-repo"
  >
    <label
      htmlFor="repo-url"
    >
      repo url:
    </label>
    <input
      name="repo-url"
      onChange={[Function]}
      style={
        Object {
          "width": 600,
        }
      }
      type="text"
      value="https://api.github.com/repos/facebook/react/issues?state=all"
    />
    <button
      onClick={[Function]}
    >
      Update
    </button>
  </div>
  <div
    id="issue-container"
  >
    <h2>
      loading issue list...
    </h2>
  </div>
</div>
`;
